### PR TITLE
Testsuite cleanup

### PR DESCRIPTION
**Pull request by HumairAK**
Resolve: https://github.com/opendatahub-io/data-science-pipelines-operator/issues/70

## Description
Clean for testing suite. Changes summarized: 

* existing tests were converted to be declarative, so you can just throw in yaml in the `testdata` folder and new cases will be auto generated
* various code refactoring, lots of redundant code was reduced. Generic client operation helpers are introduced (for example instead of having to create new functions everytime we want to deploy a new k8s resource for a test). 
* Main testing file for the controller has been pruned/cleaned up, and all helper code were moved to testutil package.

## How Has This Been Tested?
* Please do not review commits individually, review the PR holistically, and refer to the commits for progression and understanding.
 
* You can test this by creating a new case folder and running the tests locally as per usual 
* confirm existing functionality works by changing expected testdata and confirming it fails, and/or has the behaviour you expect 
* No need to spin up a live testing environment, you can test this locally 

## Merge criteria:
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: controllers/testutil/equalities.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	if !reflect.DeepEqual(expectedDep.ObjectMeta.Labels, actualDep.ObjectMeta.Labels) {
+		return false, notEqualMsg("labels")
+	}
+
+	if !reflect.DeepEqual(expectedDep.Spec.Selector, actualDep.Spec.Selector) {
+		return false, notEqualMsg("selector")
+	}
+
+	if !reflect.DeepEqual(expectedDep.Spec.Template.ObjectMeta, actualDep.Spec.Template.ObjectMeta) {
+		return false, notEqualMsg("selector")
```
Comment (harshad16) [inline controllers/testutil/equalities.go:101]: should this be ?
```suggestion
		return false, notEqualMsg("spec template")
```

File: controllers/testutil/equalities.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		}
+		// Check each env individually for a more meaningful response upon failure.
+		for i, expectedEnv := range expectedContainer.Env {
+			actualEnv := actualContainer.Env[i]
+			if !reflect.DeepEqual(expectedEnv, actualEnv) {
+				return false, notEqualMsg(fmt.Sprintf("Container Env [expected: %s, actual: %s]", expectedEnv.Name, actualEnv.Name))
+			}
+		}
+
+		if !reflect.DeepEqual(expectedContainer.Env, actualContainer.Env) {
```
Comment (harshad16) [inline controllers/testutil/equalities.go:126]: one question: if we are checking each env above part,
then is this checking for the same ?

File: controllers/testdata/declarative/case_3/expected/not_created/sample-pipeline.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: sample-pipeline-testdsp3
+    namespace: default
+    labels:
+        app: ds-pipeline-testdsp2
```
Comment (harshad16) [inline controllers/testdata/declarative/case_3/expected/not_created/sample-pipeline.yaml.tmpl:7]:
```suggestion
        app: ds-pipeline-testdsp3
```
should this be same label as app

File: controllers/testdata/declarative/case_3/expected/not_created/sample-config.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: sample-config-testdsp3
+    namespace: default
+    labels:
+        app: ds-pipeline-testdsp2
```
Comment (harshad16) [inline controllers/testdata/declarative/case_3/expected/not_created/sample-config.yaml.tmpl:7]:
```suggestion
        app: ds-pipeline-testdsp3
```
should this be same label as app?

**Review by harshad16 — COMMENTED** [review]
lgtm
just a few doubts

**Comment by HumairAK** [comment]
This PR only changes ginkgo tests, and has no changes to controller logic. 
/override ci/prow/dsp-operator-e2e

**Review by harshad16 — COMMENTED** [review]
/lgtm

thanks :+1:

**Review by harshad16 — APPROVED** [review]
thanks :+1:

**Pull request merged**